cmd/snap-confine/snap-confine.c

```diff
@@ -154,7 +174,9 @@ int main(int argc, char **argv)
 #if 0
 	setup_user_xdg_runtime_dir();
 #endif
-
+	if (snap_context != NULL) {
+		sc_context_set_environment(snap_context);
```

Comment:
Since you check for NULL in both places maybe just call this `sc_maybe_set_context_environment()` and allow it to be called with NULL.